### ISSUE TITLE
Lab 01: unable to pass test case 5

### ISSUE THREAD

**Issue by polytetralhx** [nus-cs2030/2223-s2#22]
# Summary

Unable to pass test case 5, would like to enquire if there are any hints as to what these test cases are about

## Description:

N/A

## Code Snippets

N/A

## Screenshots

![image](https://user-images.githubusercontent.com/87745260/214531634-439ab14e-1c42-44de-8c3f-8c828e6c1fc8.png)

**Comment by SimonSeeYongsheng** [comment]
If I'm not mistaken, I think the test cases are the maximum points a disc can cover?

**Comment by polytetralhx** [comment]
> If I'm not mistaken, I think the test cases are the maximum points a disc can cover?

you're right haha i just changed my code for task 5 and the output is different now, though I can't seem to figure out if i keep missing out 2 points under every test case on my algorithm. Does anyone have any idea what could be the issue?

![image](https://user-images.githubusercontent.com/87745260/214544223-7e7879f5-fb5c-43ff-9d76-27e4c21f25d8.png)

**Comment by SimonSeeYongsheng** [comment]
> > If I'm not mistaken, I think the test cases are the maximum points a disc can cover?
> 
> you're right haha i just changed my code for task 5 and the output is different now, though I can't seem to figure out if i keep missing out 2 points under some test cases on my algorithm. Does anyone have any idea what could be the issue?
> 
> ![image](https://user-images.githubusercontent.com/87745260/214544223-7e7879f5-fb5c-43ff-9d76-27e4c21f25d8.png)

Did you forget to account for the two points that are lying at the perimeter of the circle?

**Comment by 0liverkong** [comment]
by right you wont have to manually add 2 to every output if you implement the contains() function correctly (watching out for comparing doubles)

**Comment by granolaluvr69** [comment]
In the original main.jsh file, in the findMaxDiscCoverage() method, in line 23, the if condition does not account for the two points which lie on the perimeter of the disc. Just change it to distPQ <= 2.0 + epsilon && distPQ > 0 and you should be able to pass the test cases. You could also manually add the 2 points when you return maxDiscCoverage, but it might trigger the checkstyle.

**Comment by rshalinisai** [comment]
i tried that but i fail one of my test case
![Exercise 2 Test Output](https://user-images.githubusercontent.com/122519417/216827121-b4f73234-ca98-45aa-bb51-133b7ab3b84e.jpg)

**Comment by sporobolous** [comment]
> i tried that but i fail one of my test case
> ![Exercise 2 Test Output](https://user-images.githubusercontent.com/122519417/216827121-b4f73234-ca98-45aa-bb51-133b7ab3b84e.jpg)
![Uploading Screenshot 2023-02-06 at 12.38.43 AM.png…]()
same... im also facing the same issue, im not sure how to debug it lol

**Comment by bevteo** [comment]
<img width="380" alt="image" src="https://user-images.githubusercontent.com/122275359/218295579-37dd4d0d-23bf-411b-ab9d-f877dc97076f.png">

Hi, has anyone encountered this problem too. Im pretty sure my contains() method takes into consideration the points on the perimeter so I'm not really sure what went wrong.

**Comment by eesean** [comment]
Hi! I used your method and I guess I'm very close now urghh

<img width="255" alt="test case" src="https://user-images.githubusercontent.com/110977266/219825986-f22db50d-fe50-4ee3-a68b-ad098122233c.png">

Would like to ask whether:
1) Does calculation makes a difference here in getting the output (I did step by step for createUnitCircle meaning I find the angle and distance before using moveTo with the Math.PI / 2, rather than all in 1 go)? 
2) Is it right to say that the contains method similarly involves distance between 2 points being smaller than or equal to radius of circle + epsilon?

**Comment by leonardwx** [comment]
anybody has an idea to fix this issue? i tried the above methods but still doesn't work
